astropy/table/table.py

```diff
@@ -2011,48 +2009,28 @@ def _convert_string_dtype(self, in_kind, out_kind, python3_only):
 
         self._init_from_cols(newcols)
 
-    def convert_bytestring_to_unicode(self, python3_only=False):
+    def convert_bytestring_to_unicode(self):
         """
         Convert bytestring columns (dtype.kind='S') to unicode (dtype.kind='U') assuming
         ASCII encoding.
 
-        Internally this changes string columns to represent each character in the string
-        with a 4-byte UCS-4 equivalent, so it is inefficient for memory but allows Python
-        3 scripts to manipulate string arrays with natural syntax.
-
-        The ``python3_only`` parameter is provided as a convenience so that code can
-        be written in a Python 2 / 3 compatible way::
-
-          >>> t = Table.read('my_data.fits')
-          >>> t.convert_bytestring_to_unicode(python3_only=True)
-
-        Parameters
-        ----------
-        python3_only : bool
-            Only do this operation for Python 3
+        Internally this changes string columns to represent each character
+        in the string with a 4-byte UCS-4 equivalent, so it is inefficient
+        for memory but allows scripts to manipulate string arrays with
+        natural syntax.
         """
-        self._convert_string_dtype('S', 'U', python3_only)
+        self._convert_string_dtype('S', 'U')
 
-    def convert_unicode_to_bytestring(self, python3_only=False):
+    def convert_unicode_to_bytestring(self):
```

Comment:
Same deal, deprecate (and leave the documentation example of `python3_only`).  Remember there may be libraries or code that want to support both astropy 2.0 and 3.0.

Reply:
OK, this option wasn't in my mind, but is absolutely true.